hudi-common/src/main/java/org/apache/hudi/common/table/view/IncrementalTimelineSyncFileSystemView.java

```diff
@@ -181,17 +204,24 @@ private void removePendingCompactionInstant(HoodieInstant instant) throws IOExce
   }
 
   /**
-   * Remove Pending compaction instant. This is called when logcompaction is converted to delta commit,
-   * so you no longer need to track them as pending.
+   * Remove Pending compaction instant. This is called when logcompaction is converted to delta commit or the log compaction
+   * operation is failed, so it is no longer need to be tracked as pending.
    *
    * @param instant Log Compaction Instant to be removed
+   * @param isSuccessful true if log compaction operation is successful, false if the operation is failed and rollbacked.
    */
-  private void removePendingLogCompactionInstant(HoodieInstant instant) throws IOException {
-    LOG.info("Removing completed log compaction instant (" + instant + ")");
-    HoodieCompactionPlan plan = CompactionUtils.getLogCompactionPlan(metaClient, instant.getTimestamp());
-    removePendingLogCompactionOperations(CompactionUtils.getPendingCompactionOperations(instant, plan)
-        .map(instantPair -> Pair.of(instantPair.getValue().getKey(),
-            CompactionOperation.convertFromAvroRecordInstance(instantPair.getValue().getValue()))));
+  private void removePendingLogCompactionInstant(HoodieInstant instant, boolean isSuccessful) throws IOException {
+    if (isSuccessful) {
+      LOG.info("Removing completed log compaction instant (" + instant + ")");
+      HoodieCompactionPlan plan = CompactionUtils.getLogCompactionPlan(metaClient, instant.getTimestamp());
+      removePendingLogCompactionOperations(CompactionUtils.getPendingCompactionOperations(instant, plan)
+          .map(instantPair -> Pair.of(instantPair.getValue().getKey(),
+              CompactionOperation.convertFromAvroRecordInstance(instantPair.getValue().getValue()))));
+    } else {
+      LOG.info("Removing failed log compaction instant (" + instant + ")");
+      removePendingLogCompactionOperations(instant.getTimestamp());
```

Comment:
why I don't see this if else with removePendingCompactionInstant. can you help me understand please

Reply:
If a compaction instant is removed then it is not safe to do incremental file sync, because there could be removal of instants and also removal of file slices, so we fail the check in TimelineDiffHelper and the job automatically fallback on doing full sync.

Reply:
I see why we are not interested in completed compaction instants now.
but wouldn't the logic applicable to clustering commits as well then ? lets sync up f2f.

Reply:
yes i think we should apply the same to clustering commit too.

Reply:
The concern, is similar to the one raised in TimelineDiffHelper class, I explained the reasoning behind it there. 
Also, one more point to note is Compaction plans are by their nature immutable, whereas replacecommits plans can be either mutable or immutable. It depends on what UpdateStrategy implementation users are using. That means traditional flow cannot rollback compaction plans except a developer manually doing a restore, and replacecommit can be rolled back.